components/dashboard/src/projects/ProjectVariables.tsx

```diff
@@ -0,0 +1,132 @@
+import { getGitpodService } from "../service/service";
+import { ProjectContext } from "./project-context";
+import { ProjectSettingsPage } from "./ProjectSettings";
+
+export default function () {
+    const { project } = useContext(ProjectContext);
+    const [ envVars, setEnvVars ] = useState<ProjectEnvVar[]>([]);
+    const [ showAddVariableModal, setShowAddVariableModal ] = useState<boolean>(false);
+
+    const updateEnvVars = async () => {
+        if (!project) {
+            return;
+        }
+        const vars = await getGitpodService().server.getProjectEnvironmentVariables(project.id);
+        const sortedVars = vars.sort((a, b) => a.name.toLowerCase() > b.name.toLowerCase() ? 1 : -1);
+        setEnvVars(sortedVars);
+    }
+
+    useEffect(() => {
+        updateEnvVars();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [project]);
+
+    const deleteEnvVar = async (variableId: string) => {
+        await getGitpodService().server.deleteProjectEnvironmentVariable(variableId);
+        updateEnvVars();
+    }
+
+    return <ProjectSettingsPage project={project}>
+        {showAddVariableModal && <AddVariableModal project={project} onClose={() => { updateEnvVars(); setShowAddVariableModal(false); }} />}
+        <div className="mb-2 flex">
+            <div className="flex-grow">
+                <h3>Environment Variables</h3>
+                <h2 className="text-gray-500">Manage environment variables for your project.</h2>
+            </div>
+            {envVars.length > 0 && <button onClick={() => setShowAddVariableModal(true)}>Add Variable</button>}
+        </div>
+        {envVars.length === 0
+            ? <div className="bg-gray-100 dark:bg-gray-800 rounded-xl w-full py-28 flex flex-col items-center">
+                <h3 className="text-center pb-3 text-gray-500 dark:text-gray-400">No Environment Variables</h3>
+                <p className="text-center pb-6 text-gray-500 text-base w-96">Here you can set <strong>project-specific environment variables</strong> that will be visible during prebuilds and (optionally) in workspaces for this project.</p>
+                <button onClick={() => setShowAddVariableModal(true)}>New Variable</button>
+            </div>
+            : <>
+                <ItemsList>
+                    <Item header={true} className="grid grid-cols-4 items-center">
+                        <ItemField>Name</ItemField>
+                        <ItemField>Value</ItemField>
+                        <ItemField>Visible in Workspaces?</ItemField>
+                        <ItemField></ItemField>
+                    </Item>
+                    {envVars.map(variable => {
+                        return <Item className="grid grid-cols-4 items-center">
+                            <ItemField>{variable.name}</ItemField>
+                            <ItemField>****</ItemField>
+                            <ItemField>{variable.censored ? 'Hidden' : 'Visible'}</ItemField>
```

Comment:
**suggestion**: No strong feelings about this one but what do you think of using a plain `Yes`/`No` here? Alternatively, or in later iterations, we could use a checkmark icon to indicate `Yes`.

```suggestion
                            <ItemField>{variable.censored ? 'Yes' : 'No'}</ItemField>
```

Reply:
Sure, happy to go with `Yes`/`No` for this iteration.

It's just that I really like how obvious and direct words like `Hidden`/`Visible` are (i.e. no need to mentally compute affirmation or negation of the column -- it's always crystal clear what the visibility is -- while for `Yes` no you might need to re-read the column heading to be 100% certain you're not making a mistake)

Reply:
@jankeromnes Your call. We can always come back and update this if we feel this could be improved. 🏀

Reply:
(Had `Yes`/`No`, but it felt a bit awkward, so I went back to `Visibility in Workspaces` --> `Visible`/`Hidden` 🙈 happy to iterate in follow-ups though)